controllers/profiles/following.js

```diff
@@ -8,14 +8,35 @@ const {QueryTypes} = require('sequelize');
 const config = require('../../databases/config/database')[env];
 
 module.exports = async (req, res) => {
-  const id = req.userId;
-  console.log(id);
+  const userId = req.userId;
+
   let sequelize;
   if (config.use_env_variable) {
     sequelize = new Sequelize(process.env[config.use_env_variable], config);
   } else {
     sequelize = new Sequelize(config.database, config.username, config.password, config);
   }
+
+  let user_topics = await sequelize.query(
+    `select tp.topic_id from topics as tp
+    left join user_topics as utp on tp.topic_id = utp.topic_id
+    where utp.user_id = :userId`,
+    {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      }
+    }
+  );
+  let topicIds = user_topics.map((topic) => topic.topic_id);
+  const similarTopicQuery =
+    topicIds.length > 0
+      ? `ARRAY( select name from topics as tp
+        left join user_topics as utp on tp.topic_id = utp.topic_id
+        where utp.user_id = "user".user_id and tp.topic_id in (:topicIds) limit 3
+      )`
+      : 'ARRAY[]::text[]';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code block is constructing a SQL query string based on the length of `topicIds`. While this may work, it's generally not recommended to construct SQL queries in this way due to potential SQL injection vulnerabilities. Even though you're using parameterized queries (`:userId` and `:topicIds`), it's still better to avoid manually constructing SQL strings. Consider refactoring this to use Sequelize's built-in methods for conditional querying.
```diff
-  const similarTopicQuery =
-    topicIds.length > 0
-      ? `ARRAY( select name from topics as tp
-        left join user_topics as utp on tp.topic_id = utp.topic_id
-        where utp.user_id = "user".user_id and tp.topic_id in (:topicIds) limit 3
-      )`
-      : 'ARRAY[]::text[]';
+  const similarTopicQuery = sequelize.literal(
+    `ARRAY(
+      SELECT name FROM topics AS tp
+      LEFT JOIN user_topics AS utp ON tp.topic_id = utp.topic_id
+      WHERE utp.user_id = "user".user_id AND tp.topic_id IN (:topicIds)
+      LIMIT 3
+    )`
+  );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->